Client/Frontend/Widgets/BottomSheetViewController.swift

```diff
@@ -0,0 +1,248 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import SnapKit
+
+/// Presents a sheet with a child view controller of choice.
+/// On iPhones it presents as a bottom drawer style.
+/// On iPads it presents as a popup at center of the screen.
+class BottomSheetViewController: UIViewController {
+    
+    private struct UX {
+        static let HandleWidth: CGFloat = 35
+        static let HandleHeight: CGFloat = 5
+        static let HandleMargin: CGFloat = 20
+    }
+    
+    /// For landscape orientation the content view isn't full width.
+    /// A fixed constant is used instead.
+    private let maxHorizontalWidth = 400
+    private let animationDuration: TimeInterval = 0.25
+    
+    //let childViewController: UIViewController
+    
+    // MARK: - Views
+
+    let contentView = UIView().then {
+        $0.layer.cornerRadius = 12
+    }
+    
+    private let backgroundOverlayView = UIView().then {
+        $0.backgroundColor = .black
+        $0.alpha = 0
+    }
+    private let handleView = UIView().then {
+        $0.backgroundColor = .black
+        $0.alpha = 0.25
+        $0.layer.cornerRadius = UX.HandleHeight / 2
+    }
+    
+    private let closeButton = UIButton().then {
+        $0.setImage(#imageLiteral(resourceName: "close_popup").template, for: .normal)
+        $0.appearanceTintColor = .lightGray
+    }
+    
+    // MARK: - Constraint properties
+
+    /// Controls height of the content view.
+    private var yPosition: CGFloat = 0 {
+        didSet {
+            let maxY = view.frame.maxY
+            
+            func update() {
+                // Update dark blur, the more of content view go away the less dark it is.
+                backgroundOverlayView.alpha = (maxY - yPosition) / maxY
```

Comment:
Should probably check `maxY != 0` otherwise you can end up with NaN and an exception thrown when you set the frame of `contentView` below